api/client.go

```diff
@@ -507,23 +520,29 @@ func (c *Client) getSinkPathFromAgent(agentAddress string) (string, error) {
 }
 
 // starts a go routine to poll the specified file for a token
-func (c *Client) pollFileForToken(filePath string) {
+func (c *Client) pollFileForToken(filePath string, quit chan int) {
 	pollingInterval := 60 * time.Second //todo - what should this be?
 	go func() {
 		for {
-			time.Sleep(pollingInterval)
-			token, err := readAgentTokenFromFile(filePath) // todo - what to do with the error?
-			// update the client's token if it has changed and there was no error reading the file
-			if err == nil && token != c.token {
-				c.modifyLock.Lock()
-				c.token = token
-				c.modifyLock.Unlock()
+			select {
+			case <-quit:
+				quit <- 1
```

Comment:
Why write to quit?  If select takes this branch, I would simply return.

Reply:
It's a way to block [here](https://github.com/hashicorp/vault/pull/6384/files#diff-5b4f052d557bcd66021311759582f972R689) before grabbing the lock.